wezterm-gui/src/termwindow/render.rs

```diff
@@ -899,12 +1195,31 @@ impl super::TermWindow {
             );
         }
 
+        let window_buttons_at_left = self.config.window_decorations
+            == window::WindowDecorations::FANCY
+            && (self.config.fancy_window_decorations.is_left
+                || (cfg!(target_os = "macos")) && cfg!(not(windows)));
+
+        let left_padding = if window_buttons_at_left {
+            if cfg!(target_os = "macos") {
```

Comment:
can we make this something that is set in OS parameters instead of `cfg!` logic in here?